### DISCUSSION TITLE
How to inteprete the output of getConstraintState?

### DISCUSSION THREAD

**Discussion by ChengshuLi** [bulletphysics/bullet3#3370] [General]
For a fixed constraint, getConstraintState returns a vector of length 6. The documentation says they are the applied constraint forces. 

Are these the magnitudes of 6 different forces? If so, what are the directions of these forces?
Or are these two forces in 3D space (6-d in total)?
Lastly, is the unit of these values Newton?

Thanks a lot!

**Comment by erwincoumans** [comment]
The fixed constraint fixes 6 degrees of freedom, so the size is 6: 3 linear and 3 angular components.
